Level.py

```diff
@@ -74,49 +74,52 @@ def define_level_properties(self):
     Tiles methods
     """
 
-    def get_touch_auth(self, tile_type):
+    def is_authorised(self, tile_type):
         """
-        test the current tile to get authorizations
+        Test the current tile to get authorizations.
         :param tile_type: string (key of the texture)
-        :rtype: boolean
+        :rtype boolean
         """
         if tile_type == "A":
             return True
-        elif tile_type == "W":
+        else:
             self.player_path = []
             return False
 
     def get_tile_properties(self):
         """
-        find the current tile
-        :rtype: string (key of the texture)
+        Find the current tile properties.
+        :rtype string (key of the texture)
         """
-        for indx in range(self.ymax):
-            for indy in range(self.xmax):
-                horizontal_location = self.matrix[indy][indx][0] <= self.point[0] < self.matrix[indy][indx][2]
-                vertical_location = self.matrix[indy][indx][1] >= self.point[1] > self.matrix[indy][indx][3]
+        for index_x in range(self.y_max):
```

Comment:
There's still a problem here : It doesn't make sense to go loop through a range of `y_max` and call the index `index_x`.